compiler/optimizations/removeUnnecessaryAutoCopyCalls.cpp

```diff
@@ -371,11 +293,8 @@ isPrimitiveCopy(Vec<Type*>& primitiveCopyTypeSet, Type* type) {
     AggregateType* ct = toAggregateType(type);
     INT_ASSERT(ct);
 
-    // If this class defines an assignment or initCopy or destructor, it is not
-    // POD.
-    if (ct->symbol->hasFlag(FLAG_HAS_USER_DESTRUCTOR) ||
-        ct->symbol->hasFlag(FLAG_HAS_USER_INIT_COPY_FN) ||
-        ct->symbol->hasFlag(FLAG_HAS_USER_ASSIGNMENT))
+    // If we marked this class as not POD, we can't do a primitive copy.
+    if (ct->symbol->hasFlag(FLAG_NOT_POD))
```

Comment:
I've updated this comment to say "marked this type". Note the code refers to everything as "class" because that was the old name in the compiler for AggregateType.